### PR TITLE
Capture search terms in playbook analytics

### PR DESCRIPTION
**Pull request by lookupdaily**
This adds the ability to track common search terms on the playbook using our analytics tool [Plausible](https://plausible.io/docs).  

This is related to issue #1024 and was a suggestion from the content team. In order to inform future content changes (e.g. restructures), it might be useful to have some kind of analytics to track common search terms. This has two benefits:
- understanding what might be difficult to find by standard navigation
- highlighting potential content gaps in the playbook

By default Plausible strips query params from urls to count page views. We can however, use a custom script to [capture query parameters](https://plausible.io/docs/custom-query-params), and track them as page views in Plausible. 

## How it works
By creating a 'Goal' for the `/search*` url in the Plausible dashboard, we can still view the total number of visitors to the search page. When clicking on the goal, common terms are listed as pages under the 'Top pages' section. Common search terms will also be listed under general page views, but this is an easy way to view them together. 

Goals shown on the Plausible Dashboard showing 'Search' goal
Top search terms shown as page views

### Special characters
This solution currently only handles spaces in search terms, and joins them together using a '-'. It does not remove other special characters. However, special characters are shown on the plausible dashboard (see example below). It is also worth noting that search itself does not currently handle special characters. This does mean that there may be more variations of the same or similar terms though - which might create additional work when analysing the top terms.
Special characters are shown in the plausible dashboard

## What's next
We are also looking ways to understand if people are finding what they're looking for on the playbook. One suggestion is to add a button on the search page to let users let us know themselves (linked to a simple form).

### CODE REVIEW

**Review by adamhughes86 — APPROVED** [review]

File: src/_layouts/default.html [hunk 1]
```diff
@@ -1,11 +1,11 @@
 <!DOCTYPE html>
 <html lang="en-GB">
   <head>
-    <meta charset="utf-8" />
+    <meta charset="utf-8"/>
     <meta
       name="viewport"
       content="width=device-width, initial-scale=1, shrink-to-fit=no"
-    />
+/>
```
Comment (yndajas) [inline src/_layouts/default.html:8]: Is this Prettier?

File: src/_layouts/default.html [hunk 2]
```diff
@@ -55,5 +31,8 @@
       {% include page-content.html %}
     </main>
     {% include contribute.html %}
+    <script src="/assets/js/plausible.js">
+      plausible('pageview', {u: prepareUrl([])})
+    </script>
```
Comment (yndajas) [inline src/_layouts/default.html]: Does this script need to execute at the end of the file rather than in the `head`?

File: src/_webpack/plausible.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+// define the `plausible` function to manually trigger events
+window.plausible = window.plausible || function () {
+  (window.plausible.q = window.plausible.q || []).push(arguments)
+}
+
+export function prepareUrl (params) {
+  console.log('got here')
```
Comment (yndajas) [inline src/_webpack/plausible.js]: Was this `console.log` meant to be deleted?

File: src/_layouts/default.html [hunk 4]
```diff
@@ -21,8 +21,32 @@
...
+      function prepareUrl(params) {
+        const url = new URL(location.href)
+        const queryParams = new URLSearchParams(location.search)
+        let customUrl = url.protocol + "//" + url.hostname + url
+          .pathname
+          .replace(/\/$/, '')
+        for (const paramName of params) {
+          const paramValue = queryParams.get(paramName)
+          if (paramValue) 
+            customUrl = customUrl + '/' + paramValue
```
Comment (yndajas) [inline src/_layouts/default.html]: I think you fixed this in the next commit, but it might be good for it to be fixed here to keep the commit history green

```javascript
          if (paramValue) {
            customUrl = customUrl + '/' + paramValue
          }
```

File: src/_layouts/default.html [hunk 5]
```diff
@@ -31,8 +31,6 @@
       {% include page-content.html %}
     </main>
     {% include contribute.html %}
-    <script src="/assets/js/plausible.js">
-      plausible('pageview', {u: prepareUrl([])})
-    </script>
+    <script src="/assets/js/plausible.js"></script>
```
Comment (yndajas) [inline src/_layouts/default.html]: Ah - I was wondering about this in the previous commit. Does it need to be a separate commit or could it just be squashed into the last?

File: src/_webpack/plausible.js [hunk 6]
```diff
@@ -1,16 +1,21 @@
...
+}
+
+const prepareUrl = (paramsToRegister) => {
   const url = new URL(window.location.href)
   const queryParams = new URLSearchParams(window.location.search)
   let customUrl = url.protocol + '//' + url.hostname + url
     .pathname
     .replace(/\/$/, '')
-  for (const paramName of params) {
+  for (const paramName of paramsToRegister) {
```
Comment (yndajas) [inline src/_webpack/plausible.js]: Is it worth amending the `captureAllPageViews` parameter to match this?

**Review by yndajas — APPROVED** [review]
I don't know how Plausible works (trust you and Adam on that!), but have a few quick tidy up suggestions

File: src/_layouts/default.html (hunk 1 above)
Comment (lookupdaily) [reply src/_layouts/default.html:8]: It may be.. I don't think we have any project rules for html...

Comment (yndajas) [reply src/_layouts/default.html:8]: Yeah probably not - we do have a `.prettierignore` file but I can't remember what `**` means. Everything?
